LeaderboardBackend/Program.cs

```diff
@@ -67,39 +73,54 @@
 	app.UseSwaggerUI(c => c.SwaggerEndpoint("/swagger/v1/swagger.json", "LeaderboardBackend v1"));
 }
 
-app.UseHttpsRedirection();
-
 app.UseAuthentication();
 app.UseAuthorization();
 
 app.MapControllers();
 
 app.Run();
 
+#endregion
 
+#region Helpers
 
 // Add application's Database Context to the container.
-static string GetConnectionString()
+static string GetConnectionString(WebApplicationBuilder builder)
 {
+	string portVar = builder.Configuration["Db:PortVar"];
 	if (
-		!EnvReader.TryGetStringValue("POSTGRES_HOST", out var host) ||
-		!EnvReader.TryGetIntValue("POSTGRES_PORT", out var port) ||
-		!EnvReader.TryGetStringValue("POSTGRES_USER", out var user) ||
-		!EnvReader.TryGetStringValue("POSTGRES_PASSWORD", out var password) ||
-		!EnvReader.TryGetStringValue("POSTGRES_DB", out var db)
+		!EnvReader.TryGetStringValue("POSTGRES_HOST", out string host) ||
+		!EnvReader.TryGetStringValue("POSTGRES_USER", out string user) ||
+		!EnvReader.TryGetStringValue("POSTGRES_PASSWORD", out string password) ||
+		!EnvReader.TryGetStringValue("POSTGRES_DB", out string db) ||
+		!EnvReader.TryGetIntValue(portVar, out int port) 
 	)
 	{
 		throw new Exception("Database env var(s) not set. Is there a .env?");
 	}
-
 	return $"Server={host};Port={port};User Id={user};Password={password};Database={db};Include Error Detail=true";
 }
 
-static void ConfigureDbContext<T>(WebApplicationBuilder builder, bool inMemoryDb) where T : DbContext
-{
-	builder.Services.AddDbContext<T>(opt =>
+static void ConfigureDbContext<T>(WebApplicationBuilder builder, bool inMemoryDb) where T : DbContext {
+	if (inMemoryDb)
 	{
-		if (inMemoryDb) opt.UseInMemoryDatabase("LeaderboardBackend");
-		else opt.UseNpgsql(GetConnectionString()).UseSnakeCaseNamingConvention();
-	});
+		builder.Services.AddDbContext<T>(opt => opt.UseInMemoryDatabase("LeaderboardBackend"));
+	} else
+	{
+		builder.Services.AddDbContext<T>(
+			opt => {
+				opt.UseNpgsql(
+					GetConnectionString(builder)
+				).UseSnakeCaseNamingConvention();
+			}
+		);
+	}
 }
+
+#endregion
+
+#region Accessible Program class
+
+public partial class Program { }
```

Comment:
Why do we need this class for?

Reply:
This is necessary to make it a visible internal to the IntegrationTest project. It needs to be able to use this script as an entry-point for the `WebApplicationFactory`. Stolen from https://docs.microsoft.com/en-us/aspnet/core/test/integration-tests?view=aspnetcore-6.0#basic-tests-with-the-default-webapplicationfactory

Reply:
Ah, interesting.